panther_analysis_tool/main.py

```diff
@@ -434,6 +437,132 @@ def generate_hash(filename: str) -> bytes:
     return hash_bytes.digest()
 
 
+def publish_release(args: argparse.Namespace) -> Tuple[int, str]:
+    # first, ensure the appropriate github access token is set in the env
+    api_token = os.environ.get("GITHUB_TOKEN")
+    if not api_token:
+        logging.error("error: GITHUB_TOKEN env variable must be set")
+        return 1, ""
+    release_url = (
+        f"https://api.github.com/repos/{args.github_owner}/{args.github_repository}/releases"
+    )
+    # setup appropriate https headers
+    headers = {"accept": "application/vnd.github.v3+json", "Authorization": f"token {api_token}"}
+    # check this tag doesn't already exist
+    response = requests.get(release_url + f"/tags/{args.github_tag}", headers=headers)
+    if response.status_code == 200:
+        logging.error("tag already exists %s", args.github_tag)
+        return 1, ""
+    # create the release directory
+    current_time = datetime.now().isoformat(timespec="seconds").replace(":", "-")
+    release_dir = args.out if args.out != "." else f"release-{current_time}"
+    # setup and generate release assets
+    return_code = setup_release(args, release_dir, api_token)
+    if return_code != 0:
+        return return_code, ""
+    # then publish to Github
+    return_code = publish_github(args.github_tag, args.body, headers, release_url, release_dir)
+    if return_code != 0:
+        return return_code, ""
+    logging.info("draft release (%s) created in repo (%s)", args.github_tag, release_url)
+    return 0, ""
+
+
+def clone_github(
+    owner: str, repo: str, branch: str, path: str, access_token: str
+) -> Tuple[int, str]:
+    repo_url = (
+        f"https://{access_token}@github.com/{owner}/{repo}"
+        if access_token
+        else f"https://github.com/{owner}/{repo}"
+    )
```

Comment:
above you return an error if `GITHUB_TOKEN` is not specified, is this check needed? Are you thinking this could be used for unauthenticated cloning in the future for some other command?

Reply:
yea - the schema stuff also does a github clone. But... I didn't explicitly refactor it to use this instead so perhaps this isn't really needed here.

Reply:
Sounds good. Up to you then, it just seemed strange to have this check here since we verified the token had to exist earlier. But if we might refactor in the future to use this elsewhere that's fine too.